src/plugin.js

```diff
@@ -98,7 +98,7 @@ NpmInstallPlugin.prototype.preCompile = function(compilation, next) {
     );
 
     this.preCompiler = webpack(config);
-    this.preCompiler.outputFileSystem = new MemoryFS();
+    this.preCompiler.outputFileSystem = createFsFromVolume(new Volume());
```

Comment:
Please add `this.preCompiler.outputFileSystem.join = path.join.bind(path)`, we need support webpack@4